common/model-views.cpp

```diff
@@ -2787,23 +2975,43 @@ namespace rs2
                         << "Current Version: " + fw + "\nRecommended Version: " + recommended;
                     if (!fw_update_required)
                     {
-                        auto id = viewer.not_model.add_notification({ msg,
-                            RS2_LOG_SEVERITY_INFO,
-                            RS2_NOTIFICATION_CATEGORY_FIRMWARE_UPDATE_RECOMMENDED });
+                        auto n = std::make_shared<fw_update_notification_model>(
+                            msg, manager, false);
+                        viewer.not_model.add_notification(n);
 
                         fw_update_required = true;
 
-                        if (manager) viewer.not_model.attach_update_manager(id, manager);
-
-                        cleanup = [id, &viewer] {
-                            viewer.not_model.dismiss(id);
-                        };
+                        related_notifications.push_back(n);
                     }
                 }                
             }
 
             auto model = std::make_shared<subdevice_model>(dev, std::make_shared<sensor>(sub), error_message, viewer);
             subdevices.push_back(model);
+
+            if (model->supports_on_chip_calib())
+            {
+                // Make sure we don't spam calibration remainers too often:
+                time_t rawtime;
+                time(&rawtime);
+                std::string id = to_string() << configurations::viewer::last_calib_notice << "." << name.second;
+                long long last_time = config_file::instance().get_or_default(id.c_str(), (long long)0);
+                
+                std::string msg = to_string()
+                    << name.first << " (S/N " << name.second << ")";
+                auto manager = std::make_shared<on_chip_calib_manager>(viewer, model, *this, dev);
+                auto n = std::make_shared<autocalib_notification_model>(
+                    msg, manager, false);
+
+                // Recommend calibration once a week per device
+                if (rawtime - last_time < 60)
```

Comment:
The weekly remainder could create a false feeling that the device shall be re-calibrated on a weekly basis.
Setting it to a monthly periods would, imho, mitigate this

Reply:
Why monthly?